keps/sig-storage/4876-mutable-csinode-allocatable/README.md

```diff
@@ -0,0 +1,846 @@
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `k8s.io/kubernetes/pkg/kubelet`: 2024-09-24 - 51%
+- `k8s.io/kubernetes/pkg/apis/storage/validation`: 2024-09-24 - 96%
+- `k8s.io/kubernetes/pkg/volume/plugins.go`: 2024-09-24 - 27.9%
+- `k8s.io/kubernetes/pkg/volume/csi/nodeinfomanager`: 2024-09-24 - 76.6%
+
+##### Integration tests
+
+<!--
+Integration tests are contained in k8s.io/kubernetes/test/integration.
+Integration tests allow control of the configuration parameters used to start the binaries under test.
+This is different from e2e tests which do not allow configuration of parameters.
+Doing this allows testing non-default options and multiple different and potentially conflicting command line options.
+-->
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- Test that updates to `CSINode.Spec.Drivers[*].Allocatable.Count` are properly reflected when the `MutableCSINodeAllocatableCount` feature gate is enabled.
+- Test that periodic updates occur at the specified `NodeAllocatableUpdatePeriodSeconds`.
+- Test that `ResourceExhausted` errors during volume attachment trigger an immediate update of the CSINode object.
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+- Test the impact on pod scheduling when `CSINode.Spec.Drivers[*].Allocatable.Count` is updated, ensuring that pods are not scheduled to nodes with insufficient capacity.
+- Test the end-to-end workflow of updating `CSINode.Spec.Drivers[*].Allocatable.Count` using a CSI driver.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag.
+- Initial unit tests/integration tests completed and enabled.
+
+#### Beta
+
+- Allowing time for feedback (at least 2 releases between beta and GA).
+- All unit tests/integration/e2e tests completed and enabled.
+  - Validate kubelet behavior when API server rejects `CSINode` updates (older API server version).
+  - Validate CSI driver behavior with and without the `NodeAllocatableUpdatePeriodSeconds` field set.
+  - Validate scheduler behavior remains consistent regardless of whether `CSINode.Spec.Drivers[*].Allocatable.Count` is being dynamically updated or not.
```

Comment:
Can we extend this validation to cover Cluster Autoscaler behavior as well?

Reply:
Absolutely, updated Beta requirements in latest rev.